lib/api/unittest/CDataFrameAnalysisInstrumentationTest.cc

```diff
@@ -129,244 +127,253 @@ BOOST_FIXTURE_TEST_CASE(testMemoryState, ml::test::CProgramCounterClearingFixtur
-    std::stringstream output;
-    auto outputWriterFactory = [&output]() {
-        return std::make_unique<core::CJsonOutputStreamWrapper>(output);
-    };
-
-    TDataFrameUPtrTemporaryDirectoryPtrPr frameAndDirectory;
-    auto spec = test::CDataFrameAnalysisSpecificationFactory{}.outlierSpec(&frameAndDirectory);
-    api::CDataFrameAnalyzer analyzer{std::move(spec), std::move(frameAndDirectory),
-                                     std::move(outputWriterFactory)};
-
-    TDoubleVec expectedScores;
-    TDoubleVecVec expectedFeatureInfluences;
-
-    TStrVec fieldNames{"c1", "c2", "c3", "c4", "c5", ".", "."};
-    TStrVec fieldValues{"", "", "", "", "", "0", ""};
-    addOutlierTestData(fieldNames, fieldValues, analyzer, expectedScores,
-                       expectedFeatureInfluences);
-    analyzer.handleRecord(fieldNames, {"", "", "", "", "", "", "$"});
-
-    rapidjson::Document results;
-    rapidjson::ParseResult ok(results.Parse(output.str()));
-    BOOST_TEST_REQUIRE(static_cast<bool>(ok) == true);
-
-    std::ifstream schemaFileStream("testfiles/instrumentation/outlier_detection_stats.schema.json");
-    BOOST_REQUIRE_MESSAGE(schemaFileStream.is_open(), "Cannot open test file!");
-    std::string schemaJson((std::istreambuf_iterator<char>(schemaFileStream)),
-                           std::istreambuf_iterator<char>());
-    rapidjson::Document schemaDocument;
-    BOOST_REQUIRE_MESSAGE(schemaDocument.Parse(schemaJson).HasParseError() == false,
-                          "Cannot parse JSON schema!");
-    rapidjson::SchemaDocument schema(schemaDocument);
-    rapidjson::SchemaValidator validator(schema);
-
-    bool hasOutlierDetectionStats{false};
-    bool initialMemoryReport{false};
-    for (const auto& result : results.GetArray()) {
-        if (result.HasMember("analytics_memory_usage") == true &&
-            result.HasMember("outlier_detection_stats") == false) {
-            initialMemoryReport = true;
-        }
-        if (result.HasMember("outlier_detection_stats")) {
-            hasOutlierDetectionStats = true;
-            BOOST_TEST_REQUIRE(result["outlier_detection_stats"].IsObject() == true);
-            if (result["outlier_detection_stats"].Accept(validator) == false) {
-                rapidjson::StringBuffer sb;
-                validator.GetInvalidSchemaPointer().StringifyUriFragment(sb);
-                LOG_ERROR(<< "Invalid schema: " << sb.GetString());
-                LOG_ERROR(<< "Invalid keyword: " << validator.GetInvalidSchemaKeyword());
-                sb.Clear();
-                validator.GetInvalidDocumentPointer().StringifyUriFragment(sb);
-                LOG_ERROR(<< "Invalid document: " << sb.GetString());
-                BOOST_FAIL("Schema validation failed");
-            }
-        }
-    }
-    BOOST_TEST_REQUIRE(hasOutlierDetectionStats);
-    BOOST_TEST_REQUIRE(initialMemoryReport);
-}
+//
+//BOOST_FIXTURE_TEST_CASE(testTrainingRegression, ml::test::CProgramCounterClearingFixture) {
```

Comment:
There are several commented out tests here.

Reply:
These tests rely on a schema validator which, as previously discussed, was decided to be non-essential for the initial PR. I'll add an explanatory comment and come back to these time permitting.